packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -1010,45 +900,150 @@ private boolean updateMapStyle(String style) {
+    return Convert.pointToPigeon(screenLocation);
+  }
+
+  @Override
+  public @NonNull Messages.PlatformLatLng getLatLng(
+      @NonNull Messages.PlatformPoint screenCoordinate) {
+    if (googleMap == null) {
+      throw new FlutterError(
+          "GoogleMap uninitialized", "getLatLng called prior to map initialization", null);
+    }
+    LatLng latLng =
+        googleMap.getProjection().fromScreenLocation(Convert.pointFromPigeon(screenCoordinate));
+    return Convert.latLngToPigeon(latLng);
+  }
+
+  @Override
+  public @NonNull Messages.PlatformLatLngBounds getVisibleRegion() {
+    if (googleMap == null) {
+      throw new FlutterError(
+          "GoogleMap uninitialized", "getVisibleRegion called prior to map initialization", null);
+    }
+    LatLngBounds latLngBounds = googleMap.getProjection().getVisibleRegion().latLngBounds;
+    return Convert.latLngBoundsToPigeon(latLngBounds);
+  }
+
+  @Override
+  public @NonNull Double getZoomLevel() {
+    if (googleMap == null) {
+      throw new FlutterError(
+          "GoogleMap uninitialized", "getZoomLevel called prior to map initialization", null);
+    }
+    return (double) googleMap.getCameraPosition().zoom;
+  }
+
+  @Override
+  public void showInfoWindow(@NonNull String markerId) {
+    markersController.showMarkerInfoWindow(markerId);
+  }
+
+  @Override
+  public void hideInfoWindow(@NonNull String markerId) {
+    markersController.hideMarkerInfoWindow(markerId);
+  }
+
+  @NonNull
+  @Override
+  public Boolean isInfoWindowShown(@NonNull String markerId) {
+    return markersController.isInfoWindowShown(markerId);
+  }
+
+  @Override
+  public @NonNull Boolean setStyle(@NonNull String style) {
+    return updateMapStyle(style);
+  }
+
+  @Override
+  public @NonNull Boolean didLastStyleSucceed() {
+    return lastSetStyleSucceeded;
+  }
+
+  @Override
+  public void clearTileCache(@NonNull String tileOverlayId) {
+    tileOverlaysController.clearTileCache(tileOverlayId);
+  }
+
+  @Override
+  public void takeSnapshot(@NonNull Messages.Result<byte[]> result) {
+    if (googleMap == null) {
+      result.error(new FlutterError("GoogleMap uninitialized", "takeSnapshot", null));
+    } else {
+      googleMap.snapshot(
+          bitmap -> {
+            if (bitmap == null) {
+              result.error(new FlutterError("Snapshot failure", "Unable to take snapshot", null));
+            } else {
+              ByteArrayOutputStream stream = new ByteArrayOutputStream();
+              bitmap.compress(Bitmap.CompressFormat.PNG, 100, stream);
+              byte[] byteArray = stream.toByteArray();
+              bitmap.recycle();
+              result.success(byteArray);
+            }
+          });
+    }
   }
 
+  /** MapsInspectorApi implementation */
   @Override
   public @NonNull Boolean areBuildingsEnabled() {
-    return googleMap.isBuildingsEnabled();
+    return Objects.requireNonNull(googleMap).isBuildingsEnabled();
```

Comment:
These I just did requireNonNull to silence the warnings (to make the warning display in the IDE more useful) since they are integration-test only, so don't need graceful handling.